### ISSUE TITLE
Shared library and installation

### ISSUE THREAD

**Issue by remicollet** [nginx/njs#691]
Please provide configure option to allow

- installation prefix (ex: /usr)
- include directory  (ex: /usr/include/njs)
- shared library

So this can be used as all libraries
allowing unit to use it, without having to rebuild it on each version
and also reducing unit executable size (grow from ~480KB to ~1300KB with the static library)

**Comment by jirutka** [comment]
@xeioex, does njs version number reflect ABI compatibility of libnjs? For example, are all 8.1.x ABI compatible from the point of view of a program dynamically linking libnjs?

**Comment by xeioex** [comment]
@jirutka 

No, unfortunately it does not. That is why 0 is the major version.

We [added and removed some functions](https://github.com/nginx/njs/compare/0.8.0...0.8.10#diff-8c56b7f30af0933086f6b790e09f87c07c1be2efe82b6a702d8277a7a98f00b9) between 0.8.0 and 0.8.10.
